pkg/daemon/server/service/rater/helper.go

```diff
@@ -123,3 +193,102 @@ func findStartIndex(lookbackSeconds int64, counts []*TimestampedCounts) int {
 	}
 	return startIndex
 }
+
+type partitionLastSeen struct {
+	count    float64
+	seenTime int64
+}
+
+type partitionMaxDuration struct {
+	// Map to keep track of the last seen count and timestamp of each partition.
+	lastSeen map[string]partitionLastSeen
+	// Map to store the maximum duration for which the value of any partition was unchanged.
+	maxUnchangedDuration map[string]int64
+}
+
+// CalculateLookback computes the maximum duration (in seconds) for which the count of messages processed by any partition
```

Comment:
nit:  CalculateLookback computes the maximum duration (in seconds) for which the count of messages processed across all the partitions of a given vertex remain unchanged...

Just adding context that this function is to calculate per vertex